source/YugiAcademy/YuGiOh/cards/tests/test_monster_card.py

```diff
@@ -0,0 +1,111 @@
+            MonsterCard.named(name=invalid_name,
+                              race='Spellcaster',
+                              attribute='Dark',
+                              level=7,
+                              attack=2500,
+                              defense=2100,
+                              description='The ultimate wizard in terms of attack and defense.')
+        assert exception_info.message_text() == 'Name must not be blank.'
+
+
+def test_monster_card_race_must_not_be_blank():
+    for invalid_race in ['', ' ']:
+        with pytest.raises(InstanceCreationFailed) as exception_info:
+            MonsterCard.named(name='Dark Magician',
+                              race=invalid_race,
+                              attribute='Dark',
+                              level=7,
+                              attack=2500,
+                              defense=2100,
+                              description='The ultimate wizard in terms of attack and defense.')
+        assert exception_info.message_text() == 'Race must not be blank.'
+
+
+def test_monster_card_attribute_must_not_be_blank():
+    for invalid_attribute in ['', ' ']:
+        with pytest.raises(InstanceCreationFailed) as exception_info:
+            MonsterCard.named(name='Dark Magician',
+                              race='Spellcaster',
+                              attribute=invalid_attribute,
+                              level=7,
+                              attack=2500,
+                              defense=2100,
+                              description='The ultimate wizard in terms of attack and defense.')
+        assert exception_info.message_text() == 'Attribute must not be blank.'
+
+
+def test_monster_card_level_must_between_1_and_12():
+    for invalid_level in [-1, 0, 13]:
+        with pytest.raises(InstanceCreationFailed) as exception_info:
+            MonsterCard.named(name='Dark Magician',
+                              race='Spellcaster',
+                              attribute='Dark',
+                              level=invalid_level,
+                              attack=2500,
+                              defense=2100,
+                              description='The ultimate wizard in terms of attack and defense.')
+        assert exception_info.message_text() == 'Level must be between 1 and 12.'
+
+
+def test_monster_card_attack_must_be_positive():
+    with pytest.raises(InstanceCreationFailed) as exception_info:
+        MonsterCard.named(name='Dark Magician',
+                          race='Spellcaster',
+                          attribute='Dark',
+                          level=7,
+                          attack=-100,
+                          defense=2100,
+                          description='The ultimate wizard in terms of attack and defense.')
+    assert exception_info.message_text() == 'Attack must be positive.'
+
+
+def test_monster_card_defense_must_be_positive():
```

Comment:
Def can be 0.

Reply:
That's why I said positive. Perhaps, you are confused with a number being strictly positive (which means that is positive and greater than zero).

Reply:
Done!